sdk/cosmos/azure-cosmos/azure/cosmos/_cosmos_client_connection.py

```diff
@@ -1740,6 +1741,69 @@ def PatchItem(self, document_link, operations, options=None, **kwargs):
         self._UpdateSessionIfRequired(headers, result, self.last_response_headers)
         return result
 
+    def Batch(self, collection_link, batch_operations, options=None, **kwargs):
+        """Executes the given operations in transactional batch.
+
+        :param str collection_link: The link to the collection
+        :param list batch_operations: The batch of operations for the batch request.
+        :param dict options: The request options for the request.
+
+        :return:
+            The result of the batch operation.
+        :rtype:
+            dict
+
+        """
+        if options is None:
+            options = {}
+
+        path = base.GetPathFromLink(collection_link, "docs")
+        collection_id = base.GetResourceIdOrFullNameFromLink(collection_link)
+
+        batch_operations = base._format_batch_operations(batch_operations)
+
+        result, self.last_response_headers = self._Batch(
+            batch_operations,
+            path,
+            collection_id,
+            options,
+            **kwargs)
+
+        final_responses = []
+        is_error = False
+        error_status = 0
+        error_index = 0
+        for i in range(len(result)):
+            status_code = result[i].get("statusCode")
+            if status_code >= 400:
+                is_error = True
+                if status_code != 424:  # Find the operation that had the error
+                    error_status = status_code
+                    error_index = i
+                final_responses.append(models.BatchOperationError(operation=batch_operations[i],
+                                                                  operation_response=result[i],
+                                                                  error=Constants.ERROR_TRANSLATIONS.get(status_code)))
+            else:
+                final_responses.append(models.BatchOperationResponse(operation=batch_operations[i],
+                                                                     operation_response=result[i]))
+        if is_error:
+            raise exceptions.CosmosBatchOperationError(status_code=error_status,
```

Comment:
because otherwise there's a lot of information we are not being able to give to the user by just packaging it into the pre-defined HttpResponseError - otherwise we would be giving them a bunch of strings to parse through for what they're looking for, which would effectively make the error experience worse than our previous take returning a dictionary with all the information related to their operations.